### DISCUSSION TITLE
spools  N/A when printer off

### DISCUSSION THREAD

**Discussion by rasasak** [Disane87/spoolman-homeassistant#135] [General]
Hi ... 
Is strange, that spools are N/A when printer is off ... 

I havespollman addon on same PC as HASS ... so spoolmann data are always accesible ... 
But When I turn off printer, all data in this integratioin are missing ... 

shoudlnt bee always visible ?

**Comment by Disane87** [comment]
Thank you for this issue! Will investigate this.

**Comment by floyduww** [comment]
I saw this issue too as I started using this last night. 

I edited coordinator.py to have the klipper check inside it's own try/except:

```
            try:
                klipper_url = config.get(KLIPPER_URL, "")
                if klipper_url is not None and klipper_url != "":
                    klipper_active_spool: int | None = await KlipperAPI(klipper_url).get_active_spool_id()
                    if klipper_active_spool is not None:
                        for spool in spools:
                            if spool["id"] == klipper_active_spool:
                                spool["klipper_active_spool"] = True
                            else:
                                spool["klipper_active_spool"] = False
            except Exception as exception:
                print("klipper api error")
```

**Reply by Disane87** [reply]
Have fixed this in #177. Please try the latest dev if it works for you. If yes, would be nice if you would report back because I would push this fix to master if it works.

**Reply by floyduww** [reply]
Thanks for that quick update.  And the dev code does work nicely.
